src/rfq/protocols/hashflow/models.rs

```diff
@@ -0,0 +1,199 @@
+use std::{collections::HashMap, str::FromStr};
+
+use alloy::primitives::Address;
+use serde::{Deserialize, Serialize};
+use tycho_common::Bytes;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct HashflowPriceLevelsResponse {
+    pub status: String, // "success" or "fail"
+    pub levels: Option<HashMap<String, Vec<HashflowMarketMakerLevel>>>,
+    pub error: Option<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct HashflowMarketMakerLevel {
+    pub pair: HashflowPair,
+    pub levels: Vec<HashflowPriceLevel>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct HashflowPair {
+    #[serde(rename = "baseToken", deserialize_with = "deserialize_string_to_checksummed_bytes")]
+    pub base_token: Bytes,
+    #[serde(rename = "quoteToken", deserialize_with = "deserialize_string_to_checksummed_bytes")]
+    pub quote_token: Bytes,
+}
+
+fn deserialize_string_to_checksummed_bytes<'de, D>(deserializer: D) -> Result<Bytes, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    let address = Address::from_str(&s).map_err(serde::de::Error::custom)?;
+    let checksum = address.to_checksum(None);
+    let checksum_bytes = Bytes::from_str(&checksum).map_err(serde::de::Error::custom)?;
+    Ok(checksum_bytes)
+}
```

Comment:
Was actually a lifesaver to discover this tool for custom deserialization. Can immediately checksum and transform to Bytes straight from API output :) 

Not sure the error handling is the best here and if I can just right away return an RFQError, I will double-check it quickly tomorrow.